packages/workbox-build/src/lib/utils/warn-about-config.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const logHelper = require('../log-helper');
+
+/**
+ * A helper to find and log any configuration options that were set which are
+ * not valid for a given usage mode.
+ *
+ * @param {Array<String>} blacklist The names of invalid configuration options.
+ * @param {Object<String,String>} fullConfig The configuration options to check.
+ */
+module.exports = (blacklist, fullConfig) => {
+  const invalidConfig = Object.keys(fullConfig)
+    .filter((configKey) => blacklist.indexOf(configKey) !== -1);
+
+  if (invalidConfig.length > 0) {
+    logHelper.warn(`These Workbox configuration options are not valid for ` +
+      `the current mode, and have been ignored: ${invalidConfig}`);
```

Comment:
Nit: Swap "have been" with "will be" as I'm assuming this log will be at the start of the build task (whatever that will be).

Would also make sense to reword this slightly away from "These Workbox configuration options". It's more to highlight that in this context it won't be used, so maybe:

> The configuration options '${invalidConfig.join(',')}' will be ignored by `workbox-build.${methodName}`.

Reply:
Folks might end up calling the methods indirectly via the Webpack or CLI, so I don't want to put `workbox-build.` in the message. But I'll reword along those lines.